### DISCUSSION TITLE
Implementing Azure app configuration (appsettings.json) resource

### DISCUSSION THREAD

**Discussion by woter1832** [OrchardCMS/OrchardCore#10745] [Ideas]
Hi,

I've been looking at using the [Azure app configuration resource](https://docs.microsoft.com/en-us/azure/azure-app-configuration/overview) with OC. 

I wonder if anyone has looked into using this?

With the caveat that I'm not a developer, I have managed to follow some articles to enable me to create a very basic app that reads the configuration values from the app config tables, but when it comes to implementing it in OC, I don't know where to start.

However, I have identified the following files that reference `IConfigurationBuilder`:
1. src\OrchardCore\OrchardCore\Shell\ShellSettingsManager.cs
2. src\OrchardCore\OrchardCore\Shell\Configuration\ShellConfigurationSources.cs
3. src\OrchardCore\OrchardCore\Shell\Configuration\ShellsConfigurationSources.cs
4. src\OrchardCore\OrchardCore.Abstractions\Shell\Configuration\IShellConfigurationSources.cs
5. src\OrchardCore\OrchardCore.Abstractions\Shell\Configuration\IShellsConfigurationSources.cs
6. src\OrchardCore\OrchardCore.Abstractions\Shell\Configuration\ShellConfiguration.cs
7. src\OrchardCore\OrchardCore.Infrastructure\Shells.Database\Configuration\DatabaseShellConfigurationSources.cs
8. src\OrchardCore\OrchardCore.Infrastructure\Shells.Database\Configuration\DatabaseShellsSettingsSources.cs
9. src\OrchardCore\OrchardCore.Shells.Azure\Configuration\BlobShellConfigurationSources.cs
10. src\OrchardCore\OrchardCore.Shells.Azure\Configuration\BlobShellsConfigurationSources.cs
11. src\OrchardCore\OrchardCore.Shells.Azure\Configuration\BlobShellsSettingsSources.cs
12. test\OrchardCore.Tests\DisplayManagement\Decriptors\DefaultShapeTableManagerTests.cs

My guess is that I need to:
1. Modify the above files with something like:
   ```
   webBuilder.ConfigureAppConfiguration(config =>
   {
     var settings = config.Build();
     var connection = settings.GetConnectionString("AppConfig");
     config.AddAzureAppConfiguration(connection);
   });    
   ```
2. Build the modules.
3. Publish/Package the modules into my own NuGet repo.
4. Reference the modified modules in .web.csproj.

Is this roughly the right way? Any guidance, or has someone already done this?

Thanks in advance for any pointers.

**Comment by deanmarcussen** [comment]
have a look at the keyvault implementation https://docs.orchardcore.net/en/latest/docs/reference/core/KeyVault.Azure/

You are on the right track, we might take a module for it, but keyvault only needed a module because of some restrictions it places on key names (-- rather than __)

You don't need to worry about any of our shell sources / managers they manage themselves, based on the 

```
webBuilder.ConfigureAppConfiguration(config =>
{
  var settings = config.Build();
  var connection = settings.GetConnectionString("AppConfig");
  config.AddAzureAppConfiguration(connection);
});
```
